book/5-data-mgmt-exp-proc/e2e-workflow.md

```diff
@@ -0,0 +1,155 @@
+    modelexp --> modelreg
+    modelexp <---> contreg
+    modelexp ---> exptrack
+    modelreg ---> modelserv
+    contreg <--> modelserv
+    modelserv ---> modelmon
+```
+
+However, if you're just starting out in building an end-to-end workflow
+for your project, this simplified flowchart would be more suitable:
+
+```{mermaid}
+flowchart TD
+    datastorage[(Data Storage)]
+    verctl[(Version Control)]
+    devwspace[Developer Workspace]
+    modelexp(Model Experimentation)
+    modelserv[(Model Serving)]
+    datastorage <---> modelexp
+    datastorage <-.-> verctl
+    verctl --> devwspace
+    devwspace --> modelexp
+    modelexp ---> modelserv
+```
+
+This guide will focus more on the simplified flowchart. If you are
+looking for what the components represent in the typical end-to-end
+workflow but not present in the simplified one, you can refer to 
+[this section instead](e2e-workflow-adv.md).
+
+## Data Storage
+
+This pertains to storing the lifeblood of all ML/DL projects, of which 
+it is paramount that you know how and where to store data, and that you
+are able to identify raw and processed data. Raw data is defined as 
+data that is obtained from primary sources such as temperature sensors,
+while processed data is defined as data that is augmented by 
+algorithm(s) or from another ML/DL model.
+
+It is also important that you know how the product is to be designed,
+such as how data is being fed into the model, or through a
+pre-processing algorithm first. You could ingest the data either 
+locally or remotely, and remote options differ between using object 
+storage for blob-like data such as image and audio files, database 
+storage for text data, of which it may be structured (SQL, etc.) or
+unstructured (NoSQL, JSON, etc.).
+
+Understanding different types of data storage is important since the
+data that you may be given may not be suitable to be fed into the model
+directly. For example, you might be given a compressed folder of Excel
+sheets, but directly processing them before feeding into the model may
+be suboptimal when you could parallise both processes and possibly 
+reducing training time. This may also make resuming work easier since 
+there are more checkpoints between the raw data and the model itself.
+
+You can check [this section](data-mgmt.md) for more information on
+data storage options.
+
+## Version Control
+
+Most, if not all, codebases benefit from version control to reduce 
+errors by rolling back changes to a known working source. You may 
+already have been exposed to it through GitHub or GitLab, which are the
+leading platforms for source code management especially for open source
+projects. Understanding how to use version control effectively would 
+make the end-to-end workflow easier.
+
+For AI practitioners such as ourselves, there is also a subset of 
+version control that targets toward datasets, which you may read 
+[this article][data-versioning] for more information.
```

Comment:
```which you may read 
[this article][data-versioning] for more information.```

Reply:
I think the link to the article still appears as a placeholder. May wish to check.

Reply:
I've fixed this issue; it seems that there's a typo on line 155 that breaks the hyperlinks.